### PR TITLE
Login authentication

### PR DESCRIPTION
**Pull request by paymog**
In order to log in, a valid username-password combo must be entered. If
an invalid information is entered, an error message will be surfaced to
the user.

### CODE REVIEW

File: webapp/views/partials/index.jade [hunk 1]
```diff
@@ -1,8 +1,12 @@
 div(class='container')
-   h1 Welcome to #{title}
+   h1 Welcome to Lunchr
```
Comment (midfords) [inline webapp/views/partials/index.jade:2]: title variable should also be removed from webapp/routes/index.js (line 6 and 17)

Comment (paymog) [inline webapp/views/partials/index.jade:2]: Good call.

File: webapp/public/javascripts/controllers.js [hunk 2]
```diff
@@ -11,7 +11,16 @@ function MainPageController($scope, $http, $location){
...
     $scope.logIn = function() {
-        $location.path('/users');
+        $scope.errorsMessages = null;
+        $http.post('/api/users/authenticate', {email: $scope.email, password: $scope.password}).
+            success(function(data, status, headers, config){
+                $location.path('/users');
+            }).
+            error(function(data, status, headers, config){
+                $scope.errorsMessages = data;
+                $scope.email = "";
```
Comment (midfords) [inline webapp/public/javascripts/controllers.js]: I think we should just clear the password field.

Comment (paymog) [inline webapp/public/javascripts/controllers.js]: What if the username is incorrect?

**Pull request merged**